File: project-code-main1/project-code-main/app/backend/group34_backend/endpoints/country.py
import sqlite3
from json import dumps
from flask_app import app
from flask_openapi3 import Tag
from auth.auth import protected
from domain.model import Countries, ErrorResponse

countries_tag = Tag(name="countries", description="Retrieving the list of countries")


@protected
@app.get('/api/countries', responses={'200': Countries, '400': ErrorResponse}, tags=[countries_tag], description="Retrieves list of countries")
def get_countries() -> Countries or ErrorResponse:
    # Connect to database file
    conn = sqlite3.connect('group34_backend/database/stadiums.db', isolation_level=None)
    cursor = conn.cursor()

    # Execute query to select all countries
    rows = cursor.execute("SELECT DISTINCT country FROM league;").fetchall()
    
    # Loop through data and add to list
    result = []
    for item in rows:
        result.append(item[0])

    # Close connection
    cursor.close()
    conn.close()

    return dumps({"countries": result}, ensure_ascii=False)

File: project-code-main1/project-code-main/app/frontend/src/api/services/country.ts
import type { components, operations } from '../../../generated/schema';
import { api } from '../api';

export type Countries = components['schemas']['Countries'];
export type Country = components['schemas']['Country'];

/**
 * Get the list of countries
 * @returns the list of countries
 */
export const getCountries = async () => {
  try {
    const data = await api.get('/api/countries').json<Countries>();
    return data;
  } catch (error) {
    return Promise.reject(error);
  }
};
